internal/usecases/reports/get_revenue_report.go

```diff
@@ -54,95 +48,103 @@ func (useCase *GetRevenueReportUseCase) Run(ctx context.Context, startDate time.
 		return GetRevenueReportResult{}, usecases.WrapUseCaseError(usecases.GetRevenueReportId, err)
 	}
 
-	allQuoteHashes := append(peginResult.QuoteHashes, pegoutResult.QuoteHashes...)
-	penalizations, err := useCase.penalizedEventRepository.GetPenalizationsByQuoteHashes(ctx, allQuoteHashes)
+	penalizations, err := useCase.getPenalizations(ctx, peginResult, pegoutResult)
 	if err != nil {
 		return GetRevenueReportResult{}, usecases.WrapUseCaseError(usecases.GetRevenueReportId, err)
 	}
 
 	totalQuoteCallFees := entities.NewWei(0)
+	totalGasFeesCollected := entities.NewWei(0)
+	totalGasSpent := entities.NewWei(0)
 	totalPenalizations := entities.NewWei(0)
-	totalProfit := entities.NewWei(0)
 
-	for _, q := range peginResult.Quotes {
-		totalQuoteCallFees = totalQuoteCallFees.Add(totalQuoteCallFees, q.CallFee)
+	// Calculate totals from pegin quotes
+	for _, quoteWithRetained := range peginResult {
+		totalQuoteCallFees = totalQuoteCallFees.Add(totalQuoteCallFees, quoteWithRetained.Quote.CallFee)
+
+		totalGasFeesCollected = totalGasFeesCollected.Add(totalGasFeesCollected, quoteWithRetained.Quote.GasFee)
+
+		// Calculate actual gas spent
+		callForUserGasSpent := entities.NewWei(0).Mul(
+			entities.NewWei(int64(quoteWithRetained.RetainedQuote.CallForUserGasUsed)),
+			quoteWithRetained.RetainedQuote.CallForUserGasPrice,
+		)
+		registerPeginGasSpent := entities.NewWei(0).Mul(
+			entities.NewWei(int64(quoteWithRetained.RetainedQuote.RegisterPeginGasUsed)),
+			quoteWithRetained.RetainedQuote.RegisterPeginGasPrice,
+		)
+		totalGasSpent = totalGasSpent.Add(totalGasSpent, callForUserGasSpent)
+		totalGasSpent = totalGasSpent.Add(totalGasSpent, registerPeginGasSpent)
 	}
-	for _, q := range pegoutResult.Quotes {
-		totalQuoteCallFees = totalQuoteCallFees.Add(totalQuoteCallFees, q.CallFee)
+
+	// Calculate totals from pegout quotes
+	for _, pair := range pegoutResult {
+		totalQuoteCallFees = totalQuoteCallFees.Add(totalQuoteCallFees, pair.Quote.CallFee)
+
+		totalGasFeesCollected = totalGasFeesCollected.Add(totalGasFeesCollected, pair.Quote.GasFee)
+
+		// Calculate actual gas spent (RSK gas + BTC fees)
+		refundPegoutGasCost := entities.NewWei(0).Mul(
+			entities.NewWei(int64(pair.RetainedQuote.RefundPegoutGasUsed)),
+			pair.RetainedQuote.RefundPegoutGasPrice,
+		)
+		bridgeRefundGasCost := entities.NewWei(0).Mul(
+			entities.NewWei(int64(pair.RetainedQuote.BridgeRefundGasUsed)),
+			pair.RetainedQuote.BridgeRefundGasPrice,
+		)
+		totalGasSpent = totalGasSpent.Add(totalGasSpent, refundPegoutGasCost)
+		totalGasSpent = totalGasSpent.Add(totalGasSpent, bridgeRefundGasCost)
+		totalGasSpent = totalGasSpent.Add(totalGasSpent, pair.RetainedQuote.SendPegoutBtcFee)
 	}
+
 	for _, p := range penalizations {
 		totalPenalizations = totalPenalizations.Add(totalPenalizations, p.Penalty)
 	}
 
-	totalProfit = totalProfit.Sub(totalQuoteCallFees, totalPenalizations)
+	// Total profit: callFees + (gasCollected - gasSpent) - penalizations
```

Comment:
This is incorrect, the penalizations are not subtracted from the gas, they are subtracted from collateral. This will add noinse and difficulty to identify gas leaks

Reply:
If you want to return the total penalizations, return it separately, but since they are related with the collateral I recommend not to mix them with the profit calculations, as they not help to understand the profit generation at all